### ISSUE TITLE
PhpdocToCommentFixer wrongfully changes type-hinting blocks

### ISSUE THREAD

**Issue by sstok** [FriendsOfPHP/PHP-CS-Fixer#1034]
```php
        foreach ($values->fields->children() as $element) {
            /** @var \SimpleXMLElement $element */
            $fieldName = $this->getFieldName((string) $element['name']);
            $fieldConfig = $this->config->getFieldSet()->get($fieldName);
```

Is transformed to:

```php
        foreach ($values->fields->children() as $element) {
            /* @var \SimpleXMLElement $element */
            $fieldName = $this->getFieldName((string) $element['name']);
            $fieldConfig = $this->config->getFieldSet()->get($fieldName);
```

Which shouldn't happen as the phpdoc contains a `@var` hint and is preceded by a variable.

And also two other cases.

```diff
-        /** @var \NumberFormatter $formatter */
+        /* @var \NumberFormatter $formatter */
         static $formatter;
```

```diff
-/**
- * @var \Composer\Autoload\ClassLoader $loader
+/*
+ * @var \Composer\Autoload\ClassLoader
  */
``` 

This happen on the source code of https://github.com/rollerworks/RollerworksSearch I'm using the 
master version of the php-cs-fixer with composer global.

/cc @GrahamCampbell

**Comment by GrahamCampbell** [comment]
@sstok This isn't actually my fixer. :)

**Comment by ceeram** [comment]
The issue with static is a bug, the issue with foreach was intentional.
Phpdoc should come before the documented element.

```
        /** @var \SimpleXMLElement $element */
        foreach ($values->fields->children() as $element) {
            $fieldName = $this->getFieldName((string) $element['name']);
            $fieldConfig = $this->config->getFieldSet()->get($fieldName);
```

this would be the correct way, we used the upcoming PSR for phpdocs as guideline here

**Comment by ceeram** [comment]
Ill work on the issue with static

**Comment by ceeram** [comment]
I added a testcase for the static, which passes, see #1035

**Comment by ceeram** [comment]
I used 1.5 version btw. I cannot find the fixer anymore in master? @GrahamCampbell @keradus any idea if the fixer was merged/renamed in master?

**Comment by ceeram** [comment]
nvm that, my master was out of date, although git pull told me everything was up to date.

**Comment by ceeram** [comment]
The same test passes on master as well.

**Comment by ceeram** [comment]
Found a way to reproduce by using the context of original code of RollerworksSearch
Fixed now in #1035

**Comment by sstok** [comment]
Ah good to know, I will update my code then :)

**Comment by keradus** [comment]
@ceeram could you take care about that `bug with static` please?

**Comment by GrahamCampbell** [comment]
@keradus Doesn't #1035 fix that bug?

**Comment by keradus** [comment]
Didn't see that, thanks ;)
Just going notifications one by one ;)

**Comment by keradus** [comment]
Closing due to merge of #1035

**Comment by sstok** [comment]
Hmm, it seems this is still not fully resolved :cry: 

```diff
-        /** @var \Closure $hintsValue */
+        /* @var \Closure $hintsValue */
         if (!$hintsValue = $sqlWalker->getQuery()->getHint('rw_where_builder')) {
             throw new \LogicException('Missing "rw_where_builder" hint for SearchValueConversion.');
         }
```

And 


```diff
     public function __construct(ManagerRegistry $registry, array $managerNames = ['default'])
     {
-        /** @var \Doctrine\ORM\Configuration $emConfig */
+        /* @var \Doctrine\ORM\Configuration $emConfig */
         foreach ($managerNames as $managerName) {
             $emConfig = $registry->getManager($managerName)->getConfiguration();
 
             $emConfig->addCustomStringFunction(
```

**Comment by ceeram** [comment]
Ill investigate and get back to you on this @sstok

**Comment by ceeram** [comment]
although `/** @var \Doctrine\ORM\Configuration $emConfig */` has to be inside the foreach this time, as it has to immediately precede the declaration

**Comment by sstok** [comment]
I did that originally but it got changed so I tried this :)

**Comment by ceeram** [comment]
So you are saying:

```
     public function __construct(ManagerRegistry $registry, array $managerNames = ['default'])
     {
         foreach ($managerNames as $managerName) {
            /** @var \Doctrine\ORM\Configuration $emConfig */
            $emConfig = $registry->getManager($managerName)->getConfiguration();

             $emConfig->addCustomStringFunction(
```

was also incorrect according to the fixer?

**Comment by sstok** [comment]
Yep :)

**Comment by ceeram** [comment]
Ok, will investigate as well then.

**Comment by ceeram** [comment]
@keradus can you please reopen this ticket?

**Comment by sstok** [comment]
Whoops :smile: 

```
foreach ($managerNames as $managerName) {
            /** @var \Doctrine\ORM\Configuration $emConfig */
            $emConfig = $registry->getManager($managerName)->getConfiguration();
```

Is now left alone, the original was

```php
    foreach ((array) $managerNames as $managerName) {
            $emConfig = $registry->getManager($managerName)->getConfiguration();
            /** @var \Doctrine\ORM\Configuration $emConfig */
            $emConfig->addCustomStringFunction(
                'RW_SEARCH_FIELD_CONVERSION',
                'Rollerworks\Component\Search\Doctrine\Orm\Functions\SqlFieldConversion'
            );
```

Which was actually wrong, so for now only when using the if (with a value-assignment) seems to fail.

**Comment by ceeram** [comment]
Ok :)

**Comment by ceeram** [comment]
PR with fix see #1043 /cc @sstok
